Examples/Modules/qed/schwinger/analysis.py

```diff
@@ -72,23 +74,22 @@ def calculate_rate(Ex,Ey,Ez,Bx,By,Bz):
 
 def do_analysis(Ex,Ey,Ez,Bx,By,Bz):
 
-    expected_total_physical_pairs_created = dV*dt*calculate_rate(Ex,Ey,Ez,Bx,By,Bz)
+    data_set = yt.load(filename)
 
+    expected_total_physical_pairs_created = dV*dt*calculate_rate(Ex,Ey,Ez,Bx,By,Bz)
     if expected_total_physical_pairs_created < 0.01:
-        assert(not os.path.isdir(filename+"/ele_schwinger/Level_0/"))
-        ## Assert whether pairs are created or not. Is there a cleaner way to do that?
+        assert(not data_set.particle_type_counts)
```

Comment:
Uhm, `ds.particle_type_counts` should be a dictionary. I think that I would check the value of `data_set.particle_type_counts["ele_schwinger"]` and `data_set.particle_type_counts["pos_schwinger"]`

Reply:
any update on this discussion?

Reply:
Yes (sorry we discussed it on our side and didn't post an update on github).
It turns out that if there are no particles at all `particle_type_counts` returns an empty dict so this assert is correct. Using `particle_type_counts["ele_schwinger"]` would result in a KeyError.
(however if there were dumped particles of another species, we would have had to assert `particle_type_counts["ele_schwinger"] == 0`)

Reply:
This is just an analysis script, so we don't have to cover all corner cases, so it is fine to keep it this way. Otherwise, would this be a slightly more robust test?
```py
np_ele = data_set.particle_type_counts["ele_schwinger"] if \
    "ele_schwinger" in data_set.particle_type_counts.keys() else 0
np_pos = data_set.particle_type_counts["pos_schwinger"] if \
    "pos_schwinger" in data_set.particle_type_counts.keys() else 0
assert(np_ele == 0 and np_pos == 0)
```
Sorry if that's a stupid suggestion.

Reply:
I've committed your suggestion. In the end it doesn't change much but it's more explicit and I guess it's also more robust (in case `yt` changes its behavior when no particles are present).